tests/subsys/sensing/ builds with warnings on gcc12

**Describe the bug**
`tests/subsys/sensing/` builds with warnings on gcc12
This causes twister to fail on native_sim for this test as it treats warnings as errors.

It has been so since the test was introduced in #73124

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=native_sim ../tests/subsys/sensing/
3. ninja
4. See warnings

**Expected behavior**
No build warnings

**Impact**
* Build warnings
* twister failing with gcc 12 for native_sim. Note this is the only test building with warnings.

**Logs and console output**
```
FAILED: zephyr/subsys/sensing/CMakeFiles/subsys__sensing.dir/sensor_mgmt.c.obj 
ccache /usr/bin/gcc -DKERNEL -DK_HEAP_MEM_POOL_SIZE=0 -DTC_RUNID=e5e1e96a93ee4657167f100f8218e19b -D__ZEPHYR__=1 -I./subsys/sensing/include -I./twister-out/native_sim/tests/subsys/sensing/subsys.sensing/zephyr/include/generated/zephyr -I./include -I./twister-out/native_sim/tests/subsys/sensing/subsys.sensing/zephyr/include/generated -I/persistant./soc/native/inf_clock -I/persistant./boards/native/native_sim -I./scripts/native_simulator/common/src/include -I./scripts/native_simulator/native/src/include -I./subsys/rtio -I./subsys/testsuite/include -I./subsys/testsuite/ztest/include -I./drivers/sensor/bosch/bmi160/. -Wshadow -fno-strict-aliasing -Werror -Os -imacros ./twister-out/native_sim/tests/subsys/sensing/subsys.sensing/zephyr/include/generated/zephyr/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wdouble-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=./tests/subsys/sensing=CMAKE_SOURCE_DIR -fmacro-prefix-map=.=ZEPHYR_BASE -fmacro-prefix-map=/persistant/scratch/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -msse2 -mfpmath=sse -fvisibility=hidden -fno-freestanding -std=c11 -MD -MT zephyr/subsys/sensing/CMakeFiles/subsys__sensing.dir/sensor_mgmt.c.obj -MF zephyr/subsys/sensing/CMakeFiles/subsys__sensing.dir/sensor_mgmt.c.obj.d -o zephyr/subsys/sensing/CMakeFiles/subsys__sensing.dir/sensor_mgmt.c.obj -c ./subsys/sensing/sensor_mgmt.c
In file included from ./subsys/sensing/sensor_mgmt.c:6:
In function ‘sensor_later_config’,
    inlined from ‘sensing_runtime_thread’ at ./subsys/sensing/sensor_mgmt.c:230:5:
./subsys/sensing/sensor_mgmt.h:29:73: error: array subscript -1 is outside array bounds of ‘struct sensing_sensor[16777215]’ [-Werror=array-bounds]
   29 |              ({ __ASSERT(sensor >= STRUCT_SECTION_START(sensing_sensor) - 1,    \
./include/zephyr/sys/__assert.h:122:23: note: in definition of macro ‘__ASSERT’
  122 |                 if (!(test)) {                                            \
      |                       ^~~~
./subsys/sensing/sensor_mgmt.c:209:9: note: in expansion of macro ‘for_each_sensor_reverse’
  209 |         for_each_sensor_reverse(sensor) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~
In file included from ./include/zephyr/toolchain/gcc.h:98,
                 from ./include/zephyr/toolchain.h:50,
                 from ./include/zephyr/sys/__assert.h:11:
./subsys/sensing/sensor_mgmt.c: In function ‘sensing_runtime_thread’:
./include/zephyr/sys/iterable_sections.h:55:45: note: at offset -128 into object ‘_sensing_sensor_list_start’ of size [0, 2147483647]
   55 | #define TYPE_SECTION_START(secname) _CONCAT(_##secname, _list_start)
      |                                             ^
./include/zephyr/toolchain/common.h:137:26: note: in definition of macro ‘_DO_CONCAT’
  137 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
./include/zephyr/sys/iterable_sections.h:55:37: note: in expansion of macro ‘_CONCAT’
   55 | #define TYPE_SECTION_START(secname) _CONCAT(_##secname, _list_start)
      |                                     ^~~~~~~
./include/zephyr/sys/iterable_sections.h:79:21: note: in expansion of macro ‘TYPE_SECTION_START’
   79 |         extern type TYPE_SECTION_START(secname)[]
      |                     ^~~~~~~~~~~~~~~~~~
./include/zephyr/sys/iterable_sections.h:106:9: note: in expansion of macro ‘TYPE_SECTION_START_EXTERN’
  106 |         TYPE_SECTION_START_EXTERN(type, secname);               \
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
./include/zephyr/sys/iterable_sections.h:258:9: note: in expansion of macro ‘TYPE_SECTION_FOREACH’
  258 |         TYPE_SECTION_FOREACH(struct struct_type, secname, iterator)
      |         ^~~~~~~~~~~~~~~~~~~~
./include/zephyr/sys/iterable_sections.h:271:9: note: in expansion of macro ‘STRUCT_SECTION_FOREACH_ALTERNATE’
  271 |         STRUCT_SECTION_FOREACH_ALTERNATE(struct_type, struct_type, iterator)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./subsys/sensing/sensor_mgmt.h:22:9: note: in expansion of macro ‘STRUCT_SECTION_FOREACH’
   22 |         STRUCT_SECTION_FOREACH(sensing_sensor, sensor)
      |         ^~~~~~~~~~~~~~~~~~~~~~
./subsys/sensing/sensor_mgmt.c:326:9: note: in expansion of macro ‘for_each_sensor’
  326 |         for_each_sensor(sensor) {
      |         ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...): host gcc 12
 - Commit SHA or Version used: main as of now, since https://github.com/zephyrproject-rtos/zephyr/commit/a736f150f7e64a63df8d1c7fb3d5321bd2357078

**Additional context**
Related to #73846

--- a/subsys/sensing/sensor_mgmt.h
+++ b/subsys/sensing/sensor_mgmt.h
@@ -26,7 +26,7 @@ extern "C" {
 	STRUCT_SECTION_END_EXTERN(sensing_sensor);				\
 	for (struct sensing_sensor *sensor = STRUCT_SECTION_END(sensing_sensor)	\
 		- 1;								\
-	     ({ __ASSERT(sensor >= STRUCT_SECTION_START(sensing_sensor) - 1,	\
+	     ({ __ASSERT(sensor >= STRUCT_SECTION_START(sensing_sensor),	\
 		"unexpected list start location");				\
 		sensor >= STRUCT_SECTION_START(sensing_sensor); });		\
 	     sensor--)

